lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_firebase_test.rb

```diff
@@ -0,0 +1,183 @@
+require 'securerandom'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      FIREBASE_TEST_RESULT = :FIREBASE_TEST_LOG_FILE
+      FIREBASE_TEST_LOG_FILE_PATH = :FIREBASE_TEST_LOG_FILE_PATH
+    end
+
+    class AndroidFirebaseTestAction < Action
+      def self.run(params)
+        # Log in to Firebase (and validate credentials)
+        test_runner = Fastlane::FirebaseTestRunner.new(key_file: params[:key_file])
+
+        # Set up the log file and output directory
+        FileUtils.mkdir_p(params[:results_output_dir])
+        Fastlane::Actions.lane_context[:FIREBASE_TEST_LOG_FILE_PATH] = File.join(params[:results_output_dir], 'output.log')
+
+        device = Fastlane::FirebaseDevice.new(
+          model: params[:model],
+          version: params[:version],
+          locale: params[:locale],
+          orientation: params[:orientation]
+        )
+
+        result = test_runner.run_tests(
+          apk_path: params[:apk_path],
+          test_apk_path: params[:test_apk_path],
+          device: device,
+          type: params[:type]
+        )
+
+        # Download all of the outputs from the job to the local machine
+        test_runner.download_result_files(
+          result: result,
+          destination: params[:results_output_dir],
+          project_id: params[:project_id],
+          key_file_path: params[:key_file]
+        )
+
+        FastlaneCore::UI.test_failure! "Firebase Tests failed – more information can be found at #{result.more_details_url}" unless result.success?
+
+        UI.success 'Firebase Tests Complete'
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Runs the specified tests in Firebase Test Lab'
+      end
+
+      def self.details
+        description
+      end
+
+      def self.available_options
+        run_uuid = SecureRandom.uuid
+
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :project_id,
+            env_name: 'GCP_PROJECT',
+            description: 'The Project ID to test in',
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :key_file,
+            env_name: 'GOOGLE_APPLICATION_CREDENTIALS',
+            description: 'The key file used to authorize with Google Cloud',
+            type: String,
+            verify_block: proc do |value|
+              next if File.file? value
+
+              UI.user_error!("Invalid key file path: #{value}")
```

Comment:
nit: would feel shorter and more idiomatic to use:
```suggestion
              UI.user_error!("Invalid key file path: #{value}") unless File.exist?(value)
```